commercetools-models/src/main/java/io/sphere/sdk/channels/ChannelDraft.java

```diff
@@ -24,9 +24,7 @@
 @ResourceDraftValue(
         abstractResourceDraftValueClass = true,
         factoryMethods = @FactoryMethod(parameterNames = {"key"}))
-public interface ChannelDraft extends WithKey {
-    String getKey();
```

Comment:
@katmatt Do you think it's a good idea to add/leave code to adapt to IDE specific appearance features? I mean what if someone has difference IDE appearance setting?

But anyhow, in that case, I am not entirely sure where would code completion be more prominent. When the builder class is generated, in this case `ChannelDraftBuilder`, the methods `ChannelDraftBuilder#getKey` and `ChannelDraftBuilder#getCustom` are also generated for it. So there is no difference in the appearance results for code completion of the methods of `ChannelDraftBuilder`. 

![screen shot 2018-07-02 at 17 23 44](https://user-images.githubusercontent.com/9512131/42172581-b38f5d1c-7e1c-11e8-8c7c-8f941c0f9544.png)

Reply:
You're right that our builders already show the methods prominently and that it might be questionable to repeat the methods in each subtype. But it's like this for some time now and I explained the reason. And I don't see why we should change it now and don't see a lot of value in removing them.

Reply:
If it's something that is not discussable, then ok I will add them back. It's not that critical :)

But IMHO, I see value (even if it's minor one) in removing them: 
It makes the codebase a little bit less confusing for any maintainer - I don't have to wonder why overridden methods exist in a sub-interface.

It also means, that every time extending a new interface, one would have to override the new methods of the parent, which is not a common thing to do.  

(just my two cents) 😉

Reply:
I really get your reasoning, but to me consistency is important too. And we still are actively working on generating our SDKs from an external specification and that's why we want to limit what we change in the existing code. And instead concentrate our energy on our code generators. Hope this makes sense to you too and looking forward to your revert of the removed methods 😄

Reply:
I understand your point: 5d3017a